maestro-client/src/main/java/maestro/android/AndroidAppFiles.kt

```diff
@@ -38,11 +38,15 @@ object AndroidAppFiles {
     }
 
     fun getApkFile(dadb: Dadb, appId: String): File {
-        val apkPath = dadb.shell("pm list packages -f --user 0 | grep $appId | head -1")
-            .output.substringAfterLast("package:").substringBefore("=$appId")
-        apkPath.substringBefore("=$appId")
+        val apkPath = dadb.shell("pm path $appId").output.removePrefix("package:").trim()
         val dst = File.createTempFile("tmp", ".apk")
-        dadb.pull(dst, apkPath)
+        try {
+            dadb.pull(dst, apkPath)
+        } catch (e: IOException) {
+            val newApkPath = "/sdcard/$appId.apk"
+            dadb.shell("cp $apkPath $newApkPath")
+            dadb.pull(dst, newApkPath)
+        }
```

Comment:
Why this change? I'm curious to learn why `/sdcard` is used as fallback

Reply:
- I encountered a `permission denied` when pulling the APK on some of the API levels
- This is because `/data` is not accessible to pull from without the device being rooted in those cases
- However running `shell cp` and copying the files over is possible
- And `/sdcard` is accessible to pull from

See this [SO answer](https://stackoverflow.com/questions/41150038/cannot-pull-apk-from-android-device-with-adb/43157127#43157127) for example

If we add multi-API-level testing on the CI, we should add simple tests for `AndroidAppFiles` as this is the kind of things that breaks with API changes.